ui/tests/reset-password.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { test, expect } from "@playwright/test";
+import { getTotpCode, setupTotp } from "./helpers/totp";
+import { startGrafanaNewUserFlow } from "./helpers/grafana";
+import { resetIdentities } from "./helpers/kratosIdentities";
+import { getRecoveryCodeFromMailSlurp } from "./helpers/mail";
+import { USER_EMAIL, USER_PASSWORD, userPassLogin } from "./helpers/login";
+
+const USER_PASSWORD_NEW = "abcABC123!!!";
+
+test("reset password from grafana", async ({ context, page }) => {
+  resetIdentities();
+  await startGrafanaNewUserFlow(page);
+
+  await expect(page.getByText("Sign in to grafana")).toBeVisible();
+  await page.getByRole("link", { name: "Reset password" }).click();
+
+  await page.getByLabel("Email").fill(USER_EMAIL);
+  await expect(page).toHaveScreenshot({ fullPage: true, maxDiffPixels: 500 });
+  await page.getByRole("button", { name: "Reset password" }).click();
+
+  await expect(page.getByText("Enter the code you received")).toBeVisible();
+  await expect(page).toHaveScreenshot({ fullPage: true, maxDiffPixels: 500 });
+  const recoveryCode = await getRecoveryCodeFromMailSlurp(context);
+  await page.getByLabel("Recovery code").fill(recoveryCode);
+  await page.getByRole("button", { name: "Submit" }).click();
+
+  await expect(page.getByText("Reset password").first()).toBeVisible();
+  await expect(page).toHaveScreenshot({ fullPage: true, maxDiffPixels: 500 });
+  await page.getByLabel("New password", { exact: true }).fill(USER_PASSWORD_NEW);
+  await page.getByLabel("Confirm New password").fill(USER_PASSWORD_NEW);
+  await page.getByRole("button", { name: "Reset password" }).click();
+
```

Comment:
I think we have too much logic here, would it be possible to split it into smaller functions so that we have functions that each handle a single step of the flow and we re-use those functions to test the different permutations of the flow?

Reply:
Yes, agreed. Refactored this to be way shorter and more modular